Tests/DatadogTests/Datadog/RUMMonitorTests.swift

```diff
@@ -0,0 +1,81 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-2020 Datadog, Inc.
+ */
+
+import XCTest
+import UIKit
+@testable import Datadog
+
+class RUMMonitorTests: XCTestCase {
+    func testWhenCallingPublicAPI_itProcessessExpectedCommandsThrougScopes() {
+        let queue = DispatchQueue(label: #function)
+        let scope = RUMScopeMock()
+
+        let monitor = RUMMonitor(applicationScope: scope, queue: queue)
+
+        let mockView = UIViewController()
+        let mockAttributes = ["foo": "bar"]
+        let mockError = ErrorMock()
+
+        monitor.start(view: mockView, attributes: mockAttributes)
+        monitor.stop(view: mockView, attributes: mockAttributes)
+        monitor.addViewError(message: "error", error: mockError, attributes: mockAttributes)
+
+        monitor.start(resource: "/resource/1", attributes: mockAttributes)
+        monitor.stop(resource: "/resource/1", attributes: mockAttributes)
+        monitor.stop(resource: "/resource/1", withError: ErrorMock(), attributes: mockAttributes)
+
+        monitor.start(userAction: .scroll, attributes: mockAttributes)
+        monitor.stop(userAction: .scroll, attributes: mockAttributes)
+        monitor.add(userAction: .tap, attributes: mockAttributes)
+
+        queue.sync {}
```

Comment:
this feels like a hack, we don't need to know this `queue`
you could do that as well
```swift
// RUMMonitor
private let queue: DispatchQueue // implementation detail, hidden

// RUMScopeMock
func commandsExpectation(for numberOfCommands: Int) -> XCTestExpectation { ... }

// In tests
wait(for: scope.commandsExpectation(for: 9))
XCTAssertEqual(scope.recordedCommands, [...])
```

Reply:
Done 👍. I did it in the same manner as `waitAndReturn...()`  in `ServerMock`:
```swift
let scope = RUMScopeMock()
// ...
let recordedCommands = scope.waitAndReturnProcessedCommands(count: 9, timeout: 0.5)
```